### DISCUSSION TITLE
Question on base64 strings in source

### DISCUSSION THREAD

**Discussion by alj1s** [niklasvh/html2canvas#2841] [General]
Looking at the source code link on https://html2canvas.hertzen.com/ I can see there are some base64 strings that look to come from some utility libraries by the same author (css-line-break and text-segmentation) and I'd like to better understand what they are for.

We are interested in using this library but these base64 strings have been flagged as a concern for security - can someone advise and help me better understand what these are doing and why they are required

**Comment by niklasvh** [comment]
They store a trie for the line break rules as defined in https://github.com/niklasvh/css-line-break/blob/master/src/LineBreak.txt which are used to determine where it is possible to perform a line break.
